src/main/java/org/apache/maven/plugins/jmod/JModCreateMojo.java

```diff
@@ -587,6 +587,12 @@ private Commandline createJModCreateCommandLine( File resultingJModFile )
             argsFile.println( getPlatformSeparatedList( configList ) );
         }
 
+        if ( mainClass != null && !mainClass.trim().isEmpty() )
```

Comment:
No need to trim, all pom elements are being trimmed by default. If there's a StringUtils available, consider using that class.